File: fixtures/scale/ruby/large/lib/check_keypad.rb
require_relative 'keypad_encrypt'
require_relative 'keypad_decrypt'
require_relative 'corpus_additive'

def check_keypad
  corpus_additive.each do |text|
    sealed = keypad_encrypt(text, 9)
    opened = keypad_decrypt(sealed, 9)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_keypad.py
from cipherlab.ciphers.keypad import keypad_encrypt, keypad_decrypt
from cipherlab.specs.keypad_spec import keypad_spec
from cipherlab.corpus.corpus_additive import corpus_additive


def check_keypad():
    spec = keypad_spec()
    for text in corpus_additive():
        sealed = keypad_encrypt(text, spec["key"])
        opened = keypad_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "additive"
